tooling-server-configuration/src/main/java/org/jboss/eap/qe/microprofile/tooling/server/logwatch/LogWatcher.java

```diff
@@ -0,0 +1,17 @@
+package org.jboss.eap.qe.microprofile.tooling.server.logwatch;
+
+import java.util.regex.Pattern;
+
+/**
+ * Interface describing log watcher which checks log for a pattern.
+ */
+public interface LogWatcher {
+
+    /**
+     * Perform search in log or its excerpt and find if a line matches the pattern.
+     * @param pattern a pattern which will be the log line matched against
+     * @return true if log line matching pattern was found in log, false otherwise
+     */
+    boolean wasLineWithPatternLogged(final Pattern pattern);
```

Comment:
Can it be shorted? Something like `isLogged`?

Reply:
It can, but IMHO it is great addition to have such self descriptive method name (that it matches lines against pattern).

Reply:
What about `wasLineLogged` then? Parameter is also descriptive. Moreover can you overload method with `String` parameter? It might come handy to just past `.*WFLY-XYZ.*` string.

Reply:
+1 for name change. I think that you will lost the readability and IDE regex support if you place just a `String` there when you really want to pass a pattern.

Reply:
Agree. However I want two methods. `wasLineLogged(Pattern pattern)` and `wasLineLogged(String pattern)`. So we can decide if we want to always write `assert logReader.wasLineLogged(Pattern.compile("*.message.*);` or just `assert logReader.wasLineLogged(".*message.*)l`

Reply:
I know what you want. It just IMHO doesn't make a sense. Either you want to pass a `String` - in that case I suggest it will be automatically considered to be desired sub string in line which will be looked after (o you will actually call `logReader.wasLineLogged("message");`) or you want to pass a pattern in which case you will call `logReader.wasLineLogged(Pattern.compile("*.message.*);`. I don't want a masquerade where you are passing a `String` but it is actually a pattern... Are you OK with this?

Reply:
I see. Good point with a sub string. I just wanted to get rid of  `Pattern.compile`. Something like `String::matches` that would save some typing. But I am OK with what we have now,